variables.tf

```diff
@@ -93,6 +93,109 @@ variable "api_server_authorized_ip_ranges" {
   default     = null
 }
 
+variable "auto_scaler_profile_balance_similar_node_groups" {
+  description = "Enable or Disable the balance similar node groups. Defaults to false."
+  type        = bool
+  default     = false
+}
+
+variable "auto_scaler_profile_empty_bulk_delete_max" {
+  description = "The maximum number of empty nodes that can be deleted at the same time. Defaults to 10."
+  type        = number
+  default     = 10
+}
+
+variable "auto_scaler_profile_expander" {
+  description = "The expander for the node group. Possible values are `random`, `most-pods`, `least-waste`, `price`, `priority`. Defaults to `random`."
+  type        = string
+  default     = "random"
+}
+
+variable "auto_scaler_profile_max_graceful_termination_sec" {
+  description = "The maximum time in seconds for which a node can be gracefully terminated. Defaults to 600."
+  type        = number
+  default     = 600
+}
+
+variable "auto_scaler_profile_max_node_provisioning_time" {
+  description = "The maximum time in seconds for which cluster autoscaler waits for node to be provisioned. Defaults to 15m."
+  type        = string
+  default     = "15m"
+}
+
+variable "auto_scaler_profile_max_unready_nodes" {
+  description = "The maximum number of unready nodes that CA tolerates. Defaults to 3."
+  type        = number
+  default     = 3
+}
+
+variable "auto_scaler_profile_max_unready_percentage" {
+  description = "The maximum percentage of unready nodes in a node group. Defaults to 45."
+  type        = number
+  default     = 45
+}
+
+variable "auto_scaler_profile_new_pod_scale_up_delay" {
+  description = "The delay after pod starts, before it's added to calculation of pending pods. It should be used together with skip_nodes_with_local_storage. Defaults to 0s."
+  type        = string
+  default     = "0s"
```

Comment:
Do we have special reason for this `0s` default value since the [provider's default value](https://registry.terraform.io/providers/hashicorp/azurerm/latest/docs/resources/kubernetes_cluster#new_pod_scale_up_delay) is `10s`?

Reply:
I believe the provider default value might not be correct in the documentation since setting it too `10s` showed a change from `0s` too `10s` when running `terraform plan`. Also, the default of the cluster autoscaler itself is `0s`, which further supports my suspicion that the provider documentation is incorrect.

Reply:
I've tried the following `auto-scaler_profile` block and the config was idempotent:

```hcl
auto_scaler_profile {
    new_pod_scale_up_delay            = "10s"
  }
```

Would you please give us a minimum sample code that `10s` as default value would cause a drift in the generated plan? Thanks!